tests/sentry/integrations/slack/test_utils.py

```diff
@@ -103,3 +103,193 @@ def test_rate_limiting(self):
         )
         with pytest.raises(ApiRateLimitedError):
             get_channel_id(self.organization, self.integration, "@user")
+
+
+class GetChannelIdFasterTest(TestCase):
+    @with_feature("organizations:slack-use-new-lookup")
+    def setUp(self):
+        self.resp = responses.mock
+        self.resp.__enter__()
+
+        self.integration = install_slack(self.event.project.organization)
+        self.add_list_response(
+            "conversations",
+            [
+                {"name": "my-channel", "id": "m-c"},
+                {"name": "other-chann", "id": "o-c"},
+                {"name": "my-private-channel", "id": "m-p-c", "is_private": True},
+            ],
+            result_name="channels",
+        )
+        self.add_list_response(
+            "users",
+            [
+                {"name": "first-morty", "id": "m", "profile": {"display_name": "Morty"}},
+                {"name": "other-user", "id": "o-u", "profile": {"display_name": "Jimbob"}},
+                {"name": "better_morty", "id": "bm", "profile": {"display_name": "Morty"}},
+            ],
+            result_name="members",
+        )
+
+    # self.add_msg_response("other-chann", "o-c")
+    # self.add_list_response(
+    #     "users",
+    #     [
+    #         {"name": "first-morty", "id": "m", "profile": {"display_name": "Morty"}},
+    #         {"name": "other-user", "id": "o-u", "profile": {"display_name": "Jimbob"}},
+    #         {"name": "better_morty", "id": "bm", "profile": {"display_name": "Morty"}},
+    #     ],
+    #     result_name="members",
+    # )
+
+    @with_feature("organizations:slack-use-new-lookup")
+    def tearDown(self):
+        self.resp.__exit__(None, None, None)
+
+    def add_list_response(self, list_type, channels, result_name="channels"):
```

Comment:
Since this is the same as the `add_list_response` in the `GetChannelIdBotTest` class, you can share it by creating a base class and then subclassing it in these other test classes. e.g.:

```
class GetChannelIdTest(TestCase):
    def add_list_response(self, list_type, channels, result_name="channels"):
        self.resp.add(
            method=responses.GET,
            url="https://slack.com/api/%s.list" % list_type,
            status=200,
            content_type="application/json",
            body=json.dumps({"ok": "true", result_name: channels}),
        )


class GetChannelIdBotTest(GetChannelIdTest):
...
```